quantization_troubleshooting.md

```diff
@@ -0,0 +1,153 @@
+The concatenation operation can be a sensitive element in quantized models.
+This is because it involves combining multiple tensors that may have significantly different value ranges, leading to potential inaccuracies in quantization.
+
+For example, the quantization of a tensor that contains values ranging in (-1, 1) together with a tensor that ranges in (-64, 64) is less accurate than the quantization of each tensor separately.
+
+**Solution**:
+You can identify problematic "concatenation" layers using the "network editor" API or by examining the quantization parameters of the input layers to the "concatenation" layer in the quantized model.
+If you discover a concatenation layer with inputs that have notably different threshold values, consider either removing this concatenation, replacing its application to a later point in the model, or balancing its input value ranges by additional an scaling operation.
+
+
+### Pytorch model - common torch.FX errors
+When implementing quantization on a PyTorch model, MCT's initial step involves converting the model into a graph representation using `torch.fx`.
+However, `torch.fx` comes with certain common limitations, with the primary one being its requirement for the computational graph to remain static.
+
+Despite these limitations, some adjustments can be made to facilitate MCT quantization.
+
+**Solution**: (assuming you have access to the model's code)
+
+Check the `torch.fx` error, and search for an identical replacement:
+
+An `if` statement in a module's `forward` method might can be easily skipped.
+The `list()` Python method can be replaced with a concatenation operation [A, B, C].
+
+
+## Debugging Tools
+
+### Network Editor
+
+To pinpoint the possible cause of accuracy degradation in your model, it may be necessary to isolate the quantization settings of specific layers.
+However, this cannot be accomplished using MCT's main API.
+
+For instance, accuracy might suffer due to quantization of certain layer types, such as `Swish` for activations or `Conv2d` for weights.
+To address this issue and manipulate individual layers within the network, you can utilize the "Network Editor" API.
+
+**Implementation**:
+
+Using the network editor API (mct.core.network_editor) you can define a set of "rules" to apply on the network, using provided filters.
+
+Please refer to our [tutorial](https://github.com/sony/model_optimization/blob/main/tutorials/notebooks/keras/debug_tools/example_keras_network_editor.ipynb) for more details.
+
+A use-case example to understand if a certain layer causes accuracy degradation - set the output quantization bits of the layer's type to 16 bits (instead of the default 8 bits defined in the TPC).
+If the accuracy improves, then it is pointing to that layer for causing the issue.
+
+*Note:* This is mainly a debugging tool and should not be used as an API for quantizing a model for deployment (may result in an unstable behavior).
+
+
+## Advanced Quantization Methods
+
```

Comment:
rephrasing suggestion for this paragraph:

In certain instances, even if we've been thorough in following the quantization process, we face challenges like significant accuracy loss in the quantized model or its size surpassing the limitations of our deployment target (e.g., the IMX500). 

To tackle these issues, MCT offers advanced features such as Mixed Precision and GPTQ. 
These functionalities are designed to address accuracy concerns and optimize model size for smoother deployment.